asf_core_data/pipeline/mcs/process/mcs_epc_joining.py

```diff
@@ -0,0 +1,392 @@
+    merged = (
+        hps.reset_index()
+        # Join MCS records to the index-matching df on MCS index
+        .merge(top_matches, how="left", left_on="index", right_on="level_0")
+        # Then join this merged df to EPC records on EPC index
+        .merge(epcs.reset_index(), how="left", left_on="level_1", right_on="index")
+        # Drop any duplicated or unnecessary columns
+        .drop(
+            columns=[
+                "standardised_postcode_x",
+                "standardised_address_x",
+                "numeric_tokens_x",
+                "ADDRESS1",
+                "ADDRESS2",
+                "POSTTOWN",
+                "POSTCODE",
+                "index_y",
+                "postcode_y",
+                "standardised_postcode_y",
+                "numeric_tokens_y",
+                "level_0",
+            ],
+            errors="ignore",
+        )
+    )
+    merged = merged.rename(
+        columns={
+            "index_x": "original_mcs_index",
+            "level_1": "original_epc_index",
+            "postcode_x": "postcode",
+        }
+    )
+
+    if drop_epc_address:
+        merged = merged.drop(columns="standardised_address_y")
+    else:
+        merged = merged.rename(columns={"standardised_address_y": "epc_address"})
+
+    if verbose:
+
+        print("After joining:\n-----------------")
+        print("Total records:", merged.shape[0])
+        print(
+            "Number matched with EPC:",
+            merged.loc[~merged["original_epc_index"].isna()].shape[0],
+        )
+        print("\n")
+
+    return merged
+
+
+def join_mcs_epc_data(
+    hps=None, epcs=None, all_records=True, drop_epc_address=True, verbose=True
+):
+    """Produce joined MCS-EPC dataframe from "unprepared" data.
+
+    Args:
+        hps (Dataframe, optional): MCS installation records.
+        If None, records are fetched automatically. Defaults to None.
+        epcs (Dataframe, optional): EPC records. If None, records
+        are fetched automatically. Defaults to None.
+        all_records (bool, optional): Whether or not to use all matching EPC
+        records or just take one. Defaults to True.
+        drop_epc_address (bool, optional): Whether or not to drop the address
+        from the EPC records used for matching. Defaults to True.
+        verbose (bool, optional): Whether or not to print diagnostic information.
+        Defaults to True.
+
+    Returns:
+        Dataframe: Matched MCS-EPC records.
+    """
+    if hps is None:
+        print("Getting HP data...")
+        hps = get_processed_installations_data()
+
+    if epcs is None:
+        epc_version = "preprocessed" if all_records else "preprocessed_dedupl"
+        print("Getting EPC data...")
+        # TODO: replace with line that loads most recent full EPC data
+        epcs = load_s3_data(
+            bucket_name,
+            "outputs/EPC/EPC_GB_preprocessed_and_deduplicated_sample_prox.csv",
+            usecols=None,
```

Comment:
Let's resolve with using the asf-core-package.
I haven't implemented direct load from S3 but I will add that soon.

I assume it would look something like this:

```suggestion
        epcs = asf_core_data.getters.epc.epc_data.load_preprocessed_epc_data(data_path='S3', 
            version=epc_version,
            usecols=[relevant features you need for this, defined in base_config.py],
```

Reply:
I've made a new issue for making MCS work as a package and have copied over the relevant comments/suggestions